### PR TITLE
NIFI-9304 - Adding PublishGCPubSubLite and ConsumeGCPubSubLite processors

### PR DESCRIPTION
**Pull request by pvillard31**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Adding PublishGCPubSubLite and ConsumeGCPubSubLite processors

This pull request is adding two processors to interact with GCP Pub/Sub Lite. I initially thought about improving the existing processors but since it uses quite different APIs, I decided to go with new processors.

I did some extensive testing in a GCP environment and it seems to be working as expected including under high load and with concurrent tasks.

Please note that I had to use the compute engine credentials or the application credentials. If providing a service account JSON payload/file in the controller service, it seems that some gRPC calls are ignoring this authentication method. It seems to be specific to Pub/Sub Lite as I was able to use the GCS, PubSub and BigQuery processors using the service account JSON file.

Some screenshots of my tests:

Screenshot 2021-10-16 at 14 14 57

Screenshot 2021-10-16 at 14 16 52

Screenshot 2021-10-16 at 14 17 06

Screenshot 2021-10-16 at 14 18 03

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on JDK 8?
- [x] Have you verified that the full build is successful on JDK 11?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [x] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java
Comment (gresockj) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java:199]: Just to confirm, this would only be populated if someone set the ordering key by publishing external to NiFi?  I couldn't find a way to set the ordering key on the `PublishGCPubSubLite` processor, but did see it being set when publishing using `gcloud`.  Perhaps a feature for another day.

**Review by gresockj — COMMENTED** [review]
This works well, and drastically outperformed the standard `PublishGCPubSub`/`ConsumeGCPubSub` when I tried it out.

One additional note -- it appears that the gRCP API is not using the credentials provided by the controller service at all if you are not running within the Compute Engine.  If you are, either `Use Application Default Credentials` or `Use Compute Engine Credentials` will work.  I don't think this should block the PR, but it should be noted as a current limitation in the processor documentation.

**Comment by pvillard31** [comment]
Thanks for the review @gresockj ! I think I addressed all of your comments!

Note that I had some weird return line characters and used dos2unix (I recently had to setup again all of my dev env...) and it was showing all lines as modified so I rebased, squash and forced push. Sorry about that...

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java
Comment (gresockj) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java:271]: Rather than setting the actual instance variable `subscriber`, we should simply call the `getSubscriber` method and catch any exceptions.  We don't want the verification process to actually modify the class.

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java
Comment (gresockj) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java:267]: `final` arguments

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java
Comment (gresockj) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java:279]: I think this will need to be `e.getMessage()`

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java
Comment (gresockj) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java:277]: The grammatical style we've been using elsewhere is: "Create the Subscriber"

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java
Comment (gresockj) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java:272]: Here we should add a verification result with a successful outcome to show that the subscriber was able to be created.

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java
Comment (gresockj) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java:80]: I think the important part to add is that it only works on a Compute Engine instance, because if you try "Use Application Default Credentials" without running on Compute, it still fails.

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/PublishGCPubSubLite.java
Comment (gresockj) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/PublishGCPubSubLite.java:310]: See same notes as in `ConsumeGCPubSubLite`

**Review by gresockj — CHANGES_REQUESTED** [review]
Thanks for the updates, @pvillard31!  Just a few more items left.

**Review by gresockj — COMMENTED** [review]
Great, that addresses all my comments, @pvillard31.  I plan to run this with the config verification updates, and if all looks good, I'll be a +1.

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/PublishGCPubSubLite.java
Comment (gresockj) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/PublishGCPubSubLite.java:240]: This can be streamlined a little by declaring the builder as a final local variable, and then adding the orderingKey if it's non-null.

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/PublishGCPubSubLite.java
Comment (gresockj) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/PublishGCPubSubLite.java:228]: Can be final, also missing a space

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/PublishGCPubSubLite.java
Comment (gresockj) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/PublishGCPubSubLite.java:107]: Perhaps `gcp-ordering-key` instead of `gcp-ordering_key`

**Review by gresockj — CHANGES_REQUESTED** [review]
Ok, config verification looks good, but I detected a couple minor changes needed for the new ORDERING_KEY support on the publisher.

**Comment by pvillard31** [comment]
Thanks @gresockj , good points, I think this should be OK now.

**Review by gresockj — APPROVED** [review]
Thanks for these last updates, @pvillard31!  I've run this again on Compute Engine, and it all works as expected now.  +1 from me, though I am going to wait before merging until after the 1.15.0 release.

**Pull request closed without merging**
